src/app/views/login/SignInController.test.js

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { LoginController } from "./SignInController";
+import renderer from "react-test-renderer";
+import { mount, shallow } from "enzyme";
+
+jest.mock("../../utilities/notifications.js", () => {
+    return {
+        add: function() {
+            // do nothing
+        }
+    };
+});
+
+jest.mock("../../utilities/websocket", () => {
+    return {
+        send: function() {
+            // do nothing
+        }
+    };
+});
+
+jest.mock("../../utilities/api-clients/user.js", () => {
+    return {
+        setUserState: function() {
+            // do nothing
+        },
+        logOut: function() {
+            // do nothing
+        }
+    };
+});
+jest.mock("../../utilities/redirectToMainScreen.js", () => {
+    return {
+        redirectToMainScreen: function() {
+            // do nothing
+        }
+    };
+});
+
+test("Check redirect doesn't work if not authenticated", () => {
+    const props = {
+        dispatch: function() {},
+        rootPath: "/florence",
+        isAuthenticated: false
+    };
+    const component = mount(<LoginController {...props} />);
+    expect(component.find("h1").text()).toBe("Log in to your Florence account");
+});
+
+test("Does password change form appear on state change", () => {
+    const props = {
+        dispatch: function() {},
+        rootPath: "/florence",
+        isAuthenticated: false
+    };
+    const component = mount(<LoginController {...props} />);
+    expect(component.find(".modal__overlay").length).toBe(0);
```

Comment:
Is this how tests are generally written in Florence? Asserting on a class that's used for styling couples your test to the styling mechanism — eg. if the class name was changed due to a refactor it would break this unrelated test, making it fragile.

It's better to use selectors that are either:

- something the user will see (eg. title of the modal)
- explicitly for tests (eg. a `data-testid="modal"`) attribute

This is a good article on this: https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change

Reply:
This was an interesting article and think will implement in one of the other PRs as discussed in standup.